more various l10n updates (4)

tijuca opened the pull request:
Initially I wanted to work on other l10n parts, but after updating the source I was required to also update my local database.
Thus I saw the menu / UI wasn't already completely translated and so I reworked this part.

So far possible I tested the various cases, but some things I wasn't able to see in real action (like some database interaction errors). So it might be possible that some small parts of strings aren't fully correct adjusted to l10n. Please kindly correct any outstanding issues here. Thanks!

Finally I updated the message catalogue again for the previous newly added strings and also for the strings I've added.

tijuca committed:
l10n: fix usage of ._() function within li element

Localization of strings with variables inside doesn't work, splitting of
the existing string into two independent strings.

tijuca committed:
l10n: authentication-methods add some more possible strings

Add the list entries of possible new authentification methods to
translatable strings.

tijuca committed:
l10n: database upgrade adding strings from dialogs

Make the user guidance through a rquired upgrade of the database
translatable.

tijuca committed:
l10n: database interactions adding more possible strings

The content of functions/classes/class.Install.php is used on various
database actions. Turning more strings into l10n strings so translation
is more generally done for interaction with any database interaction in
phpIPAM.

GaryAllan commented:
> So far possible I tested the various cases, but some things I wasn't able to see in real action (like some database interaction errors). So it might be possible that some small parts of strings aren't fully correct adjusted to l10n. Please kindly correct any outstanding issues here. 

We `setlocale()` the language from the database ('settings' table for global and 'users' for per-user). During install and database connectivity issues this isn't set because the database isn't available to read the values.

A sane thing to do would be to `setlocale()` using the LC_ALL environment variable when the database isn't available.

https://github.com/phpipam/phpipam/blob/bd2cb4463eebd3474a196cb073948708aab22533/app/login/index.php#L116-L124

https://github.com/phpipam/phpipam/blob/bd2cb4463eebd3474a196cb073948708aab22533/functions/classes/class.User.php#L469-L476

tijuca commented:
Hi Gary,
so you mean e.g. to initialize phpipam at startup with the LC_ALL environment that would be directly overridden with the saved locale from the database to use for the user? Sound reasonable to my.

Otherwise I can change things (l10n marked strings) if you prefer that.
I'm not a PHP coder so can't provide anything useful on that side.

GaryAllan commented:
Hello @tijuca 
Can you rebase this PR against master?

06d0345affaa428a6843ddd427440ccfed01d5f1, 2677ffa1649e1cb4a4f309aa93be929bb2fce48a and 61674a3bd53056b60bc5dc188153f6aa6e50f849 should make things easier for you.

tijuca committed:
l10n: Update German translations

Update the po file and also the binary message catalog for German after
reworking various source files.

tijuca commented:
Hi Gary, I've found some time to do the rebase and afterwards some updates to German po file.
Looks better now especially the l10n strings in the user and admin menu. No untranslated strings here any more.
